testkit-backend/backend.go

```diff
@@ -476,9 +484,14 @@ func (b *backend) handleRequest(req map[string]interface{}) {
 			b.writeError(err)
 			return
 		}
+		keys, err := result.Keys()
+		if err != nil {
+			b.writeError(err)
+			return
+		}
 		idKey := b.nextId()
 		b.results[idKey] = result
-		b.writeResponse("Result", map[string]interface{}{"id": idKey})
+		b.writeResponse("Result", map[string]interface{}{"id": idKey, "keys": keys})
```

Comment:
I could also completely rewrite the `result#Keys` implementation so it stores the keys right away and doesn't go through the stream lookup to get the keys when asked...

Reply:
(applies below too)